src/sensors/mpu9250sensor.cpp

```diff
@@ -382,4 +366,101 @@ void MPU9250Sensor::startCalibration(int calibrationType) {
     m_Logger.debug("Saved the calibration data");
 
     m_Logger.info("Calibration data gathered");
+    // fifo will certainly have overflown due to magnetometer calibration, reset it.
+    imu.resetFIFO();
+}
+
+void MPU9250Sensor::parseMagData(int16_t data[3]) {
+    // reading *little* endian int16
+    Mxyz[0] = (float)data[0];
+    Mxyz[1] = (float)data[1];
+    Mxyz[2] = -(float)data[2];
+
+    float temp[3];
+
+    //apply offsets and scale factors from Magneto
+    for (unsigned i = 0; i < 3; i++) {
+        temp[i] = (Mxyz[i] - m_Calibration.M_B[i]);
+        #if useFullCalibrationMatrix == true
+            Mxyz[i] = m_Calibration.M_Ainv[i][0] * temp[0] + m_Calibration.M_Ainv[i][1] * temp[1] + m_Calibration.M_Ainv[i][2] * temp[2];
+        #else
+            Mxyz[i] = temp[i];
+        #endif
+    }
 }
+
+void MPU9250Sensor::parseAccelData(int16_t data[3]) {
+    // reading big endian int16
+    Axyz[0] = (float)data[0];
+    Axyz[1] = (float)data[1];
+    Axyz[2] = (float)data[2];
+
+    float temp[3];
+
+    //apply offsets (bias) and scale factors from Magneto
+    for (unsigned i = 0; i < 3; i++) {
+        temp[i] = (Axyz[i] - m_Calibration.A_B[i]);
+        #if useFullCalibrationMatrix == true
+            Axyz[i] = m_Calibration.A_Ainv[i][0] * temp[0] + m_Calibration.A_Ainv[i][1] * temp[1] + m_Calibration.A_Ainv[i][2] * temp[2];
+        #else
+            Axyz[i] = temp[i];
+        #endif
+    }
+}
+
+// TODO: refactor so that calibration/conversion to float is only done in one place.
+void MPU9250Sensor::parseGyroData(int16_t data[3]) {
+    // reading big endian int16
+    Gxyz[0] = ((float)data[0] - m_Calibration.G_off[0]) * gscale; //250 LSB(d/s) default to radians/s
+    Gxyz[1] = ((float)data[1] - m_Calibration.G_off[1]) * gscale;
+    Gxyz[2] = ((float)data[2] - m_Calibration.G_off[2]) * gscale;
+}
+
+// really just an implementation detail of getNextSample...
+void MPU9250Sensor::swapFifoData(union fifo_sample_raw* sample) {
```

Comment:
I should consider moving the internals of swapFifoData directly into getNextSample